main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeTravelSpeedValidatorTest.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mobilitydata.gtfsvalidator.validator.StopTimeTravelSpeedValidator.getSpeedKphBetweenStops;
+import static org.mobilitydata.gtfsvalidator.validator.StopTimeTravelSpeedValidator.getStopLatLng;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.geometry.S2LatLng;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsRoute;
+import org.mobilitydata.gtfsvalidator.table.GtfsRouteTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsRouteType;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+import org.mobilitydata.gtfsvalidator.util.S2Earth;
+import org.mobilitydata.gtfsvalidator.validator.StopTimeTravelSpeedValidator.FastTravelBetweenConsecutiveStopsNotice;
+import org.mobilitydata.gtfsvalidator.validator.StopTimeTravelSpeedValidator.FastTravelBetweenFarStopsNotice;
+
+@RunWith(JUnit4.class)
+public final class StopTimeTravelSpeedValidatorTest {
```

Comment:
Suggest adding test case for legacy GTFS without timepoint column that may have missing arrival and departure times for non-timepoint records:

|  stop_sequence | arrival_time | departure_time |
|  ------ | ------ | ------ | 
1 | 00:00:00 | 00:00:00  |  
2 |  |  |  
3 |  |  |  
4 | 00:10:00 | 00:10:00  |

Reply:
>By the way, timepoint column is optional. If arrival_time and departure_time are provided, there still may be no timepoint column.

Yes, that's correct.

>What is legacy GTFS that you mentioned?

Basically GTFS without the timepoint column, which has two flavors:
1. The above table, where non-timepoint records omitted arrival and departure times. This format strictly conformed to the original GTFS spec before the timepoint field was added.
2. Arrival and departure times were provided for all records in stop_times.txt. This format did not strictly conform to the original GTFS spec (which said that times should only be provided for timepoints), but it became a common practice because agencies wanted different consumers to show the same scheduled times for stops (vs. interpolating a scheduled time themselves, which might differ between consumer implementations). The timepoint field was added to allow producers to provide times for all stops, but also explicitly mark certain stops as timepoints. So ideally these types of datasets should eventually be phased out of existence by the agency adding the timepoint column - I don't know what the current state of the world is for this now.

See this thread with more examples and details - https://github.com/google/transit/issues/61#issuecomment-702329132 - which also discusses potential changes to the spec language to make the expected use of the field clearer.

Reply:
Thanks a lot for explanation, this is very nice to know!